clang/test/Sema/warn-cast-function-type.c

```diff
@@ -19,8 +19,12 @@ f5 *e;
 f6 *f;
 f7 *g;
 
+enum E : long;
```

Comment:
Another test that would be helpful would be a test not using a fixed underlying type but still produces a type compatible with `long`, as in:
```
enum E {
  Big = __LONG_MAX__
};
```

Reply:
Good idea, done!